scene/2d/physics_body_2d.cpp

```diff
@@ -1198,7 +1198,7 @@ Vector2 KinematicBody2D::move_and_slide(const Vector2 &p_linear_velocity, const
 						revert_motion();
 						return Vector2();
 					}
-				} else if (get_collision_normal().dot(p_floor_direction) < Math::cos(Math::deg2rad((float)45))) { //ceiling
+				} else if (get_collision_normal().dot(p_floor_direction) <= -Math::cos(Math::deg2rad(p_floor_max_angle))) { //ceiling
```

Comment:
@tagcup What would you do about this though? 0.785398 isn't a really intuitive default value...
![image](https://cloud.githubusercontent.com/assets/5276727/24664195/b410b1e8-1962-11e7-9420-27516dab6bab.png)